src/client/dfuse/dfuse_pool.c

```diff
@@ -0,0 +1,111 @@
+/**
+ * (C) Copyright 2016-2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#include "dfuse_common.h"
+#include "dfuse.h"
+#include "daos_fs.h"
+#include "daos_api.h"
+
+/* Lookup a pool */
+bool
+dfuse_pool_lookup(fuse_req_t req, struct dfuse_inode_entry *parent,
+		  const char *name)
+{
+	struct dfuse_projection_info	*fs_handle = fuse_req_userdata(req);
+	struct dfuse_info		*dfuse_info = fs_handle->dfuse_info;
+	struct dfuse_inode_entry	*ie = NULL;
+	struct dfuse_dfs		*dfs = NULL;
+	uuid_t				pool_uuid;
+	int rc;
+
+	/* This code is only supposed to support one level of directory descent
+	 * so check that the lookup is relative to the root of the sub-tree,
+	 * and abort if not.
+	 */
+	D_ASSERT(parent->ie_stat.st_ino == parent->ie_dfs->dffs_root);
+
+	if (uuid_parse(name, pool_uuid) < 0) {
+		DFUSE_LOG_ERROR("Invalid container uuid");
+		D_GOTO(err, rc = ENOENT);
+	}
+
+	D_ALLOC_PTR(dfs);
+	if (!dfs) {
+		D_GOTO(err, rc = ENOMEM);
+	}
+	strncpy(dfs->dffs_pool, name, NAME_MAX);
+
+	rc = daos_pool_connect(pool_uuid, dfuse_info->dfi_dfd.group,
+			       dfuse_info->dfi_dfd.svcl, DAOS_PC_RW,
+			       &dfs->dffs_poh, &dfs->dffs_pool_info,
+			       NULL);
+	if (rc != -DER_SUCCESS) {
+		DFUSE_LOG_ERROR("daos_pool_connect() failed: (%d)",
+				rc);
+		D_GOTO(err, 0);
+	}
+
+	D_ALLOC_PTR(ie);
+	if (!ie) {
+		D_GOTO(close, rc = ENOMEM);
+	}
+
+	ie->ie_parent = parent->ie_stat.st_ino;
+	strncpy(ie->ie_name, name, NAME_MAX);
+
+	atomic_fetch_add(&ie->ie_ref, 1);
+	ie->ie_dfs = dfs;
+
+	ie->ie_stat.st_uid = geteuid();
+	ie->ie_stat.st_gid = getegid();
+	ie->ie_stat.st_nlink = 1;
+	ie->ie_stat.st_mode = S_IFDIR | 0755;
+
+	ie->ie_stat.st_atim.tv_sec = time(NULL);
+	ie->ie_stat.st_mtim.tv_sec = ie->ie_stat.st_mtim.tv_sec;
```

Comment:
Is that me or the two variables are the same (i.e. a = a)?

Reply:
Typo, fixed.

Reply:
These aren't actually required and it's not possible to read them so I've removed these entries entirely.